extensions/machine-learning/src/packageManagement/packageManager.ts

```diff
@@ -196,23 +199,25 @@ export class PackageManager {
 	}
 
 	private async installPipPackage(requirementFilePath: string): Promise<string> {
-		let cmd = `"${this.pythonExecutable}" -m pip install -r "${requirementFilePath}"`;
+		let pythonExecutable = await this.getPythonExecutable();
+		let cmd = `"${pythonExecutable}" -m pip install -r "${requirementFilePath}"`;
 		return await this._processService.executeBufferedCommand(cmd, this._outputChannel);
 	}
 
 	private async installRPackage(model: PackageConfigModel): Promise<string> {
 		let output = '';
 		let cmd = '';
+		let rExecutable = await this.getRExecutable();
 		if (model.downloadUrl) {
 			const packageFile = utils.getPackageFilePath(this._rootFolder, model.fileName || model.name);
 			const packageExist = await utils.exists(packageFile);
 			if (!packageExist) {
 				await this._httpClient.download(model.downloadUrl, packageFile, this._outputChannel);
 			}
-			cmd = `"${this._rExecutable}" CMD INSTALL ${packageFile}`;
+			cmd = `"${rExecutable}" CMD INSTALL ${packageFile}`;
 			output = await this._processService.executeBufferedCommand(cmd, this._outputChannel);
```

Comment:
Long term  - I think the execute command code is something needed by so many extensions (I know 3 for sure ML/Proj) we should work with Alan to see if we can have some common utils for this hat extns can share.

Reply:
sure, there are so many modules added here which will be useful to move to share location

---
In reply to: [425310850](https://github.com/microsoft/azuredatastudio/pull/10377#discussion_r425310850) [](ancestors = 425310850)